UI/src/app/services/upload.service.ts

```diff
@@ -28,24 +29,40 @@ export class UploadService {
 
   onFileChange(event: any): void {
     const files = event.target.files;
-    const dotPosition = files[0].name.lastIndexOf('.');
+    const dotPosition = files[ 0 ].name.lastIndexOf('.');
     this.uploadSchema(files).subscribe(res => {
       this.snackbar.open(
         'Success file upload',
         ' DISMISS ',
-        {duration: 3000}
+        { duration: 3000 }
       );
       this.bridgeService.resetAllMappings();
       this.dataService.prepareTables(res, 'source');
-      this.dataService.saveReportName(files[0].name.slice(0, dotPosition), 'report')
+      this.dataService.saveReportName(files[ 0 ].name.slice(0, dotPosition), 'report')
       this.bridgeService.saveAndLoadSchema$.next();
     });
   }
 
+  onMappingChange(event: any): void {
+    const file = event.target.files[ 0 ];
+    const fileReader: FileReader = new FileReader();
+    let content;
+    const self = this;
+    fileReader.onloadend = () => {
+      content = fileReader.result;
+      const loadedConfig = JSON.parse(content);
+      const resultConfig = new Configuration();
+      Object.keys(loadedConfig).forEach(key => resultConfig[key] = loadedConfig[key]);
+      self.bridgeService.applyConfiguration(resultConfig);
```

Comment:
inside of arrow functions you can use context from outside, because arrow functions has no it's own.
get rid of self

Reply:
Done